### PR TITLE
Add implementation for hint on assert_nn (math.cairo)

### PR DESCRIPTION
**Pull request by pefontana**
# Add implementation for hint on assert_nn (math.cairo)

Add implementation for hint on assert_nn cairo (math.cairo)
https://github.com/starkware-libs/cairo-lang/blob/167b28bcd940fd25ea3816204fa882a0b0a49603/src/starkware/cairo/common/math.cairo#L39

## Checklist
- [ ] Linked to Github Issue
- [x] Unit tests added
- [x] Integration tests added.
- [ ] This change requires new documentation.
  - [ ] Documentation has been added/updated.

### CODE REVIEW

File: src/vm/hints/hint_utils.rs [hunk 1]
```diff
@@ -275,3 +276,62 @@ pub fn assert_le_felt(
...
+                                return Err(VirtualMachineError::ValueOutOfRange(a.clone()));
+                            }
+                        }
+                    }
+                }
+            }
+            Err(VirtualMachineError::NoRangeCheckBuiltin)
+        }
+        _ => Err(VirtualMachineError::FailedToGetIds),
+    }
```
Comment (Oppen) [inline src/vm/hints/hint_utils.rs:336]: We can probably reduce nesting here. E.g.:
```
    let maybe_rel_a = if let Ok(Some(maybe_rel_a)) = vm.memory.get(&a_addr) {
        maybe_rel_a
    } else {
        return Err(VirtualMachineError::FailedToGetIds);
    }
```

Then the loop could only contain the linear search and then just use the result after.

Comment (pefontana) [reply src/vm/hints/hint_utils.rs]: Thanks for the advice pelito.
I implemented this modification in this [commit](https://github.com/lambdaclass/cleopatra_cairo/pull/243/commits/8b6db26679cb9e96964a79076eed16d80981e496)

File: src/vm/hints/hint_utils.rs [hunk 2]
```diff
@@ -303,35 +303,41 @@ pub fn assert_nn(
...
-                                return Ok(());
-                            } else {
-                                return Err(VirtualMachineError::ValueOutOfRange(a.clone()));
-                            }
-                        }
-                    }
-                }
+
+            // assert 0 <= ids.a % PRIME < range_check_builtin.bound
+            if bigint!(0) <= a.mod_floor(&vm.prime)
```
Comment (Oppen) [inline src/vm/hints/hint_utils.rs:333]:
```suggestion
            if !a.mod_floor(&vm.prime).is_positive()
```

Comment (pefontana) [reply src/vm/hints/hint_utils.rs]: Done!
`if a.mod_floor(&vm.prime).is_positive()`

**Review by Oppen — APPROVED** [review]

**Pull request merged**